dashboards-observability/public/components/synthetics/home_view.tsx

```diff
@@ -246,51 +382,202 @@ export const SyntheticHomeTab = (props: LogHistoryProps) => {
+                    color: 'red',
+                  },
+                  // opacity: opacityValue
+                },
+              ]}
+              layout={{
+                height: 750,
+                width: 1200,
+                geo: {
+                  scope: 'world',
+                  showcountries: true,
+                  projection: {
+                    type: 'natural earth',
+                  },
+                  showland: true,
+                  landcolor: 'rgb(243,243,243)',
+                  countrycolor: 'rgb(204,204,204)',
+                  fitbounds: 'locations',
+                },
+              }}
+            />
+          </EuiFlexItem>
+          <EuiFlexItem>
+            <Plt
+              data={[
+                {
+                  values: pieChartLogs[0],
+                  labels: pieChartLogs[1],
+                  type: 'pie',
+                },
+              ]}
               layout={{
                 height: 250,
-                width: 250
+                width: 250,
               }}
             />
           </EuiFlexItem>
         </EuiFlexGroup>
       </EuiPageContent>
       <EuiSpacer size="xl" />
-      <EuiFlexGroup>
+      <>
         <EuiFlexItem grow={false} align-items="center">
-          <EuiText>
-            Synthetics History
-          </EuiText>
+          <EuiText>Synthetics History</EuiText>
         </EuiFlexItem>
         <EuiFlexItem grow={false}>
           <EuiFacetGroup layout="horizontal" gutterSize="none">
-            <EuiFacetButton onClick={() => {pplServiceRequestor(allLogsQuery + filterVal + timeVal, "jsonData", setHistoryLogs)
-                                            setHistoryQuery(allLogsQuery + filterVal + timeVal)}}>
+            <EuiFacetButton
+              onClick={() => {
+                setHistoryQuery(allLogsQuery());
+              }}
+            >
               All
             </EuiFacetButton>
-            <EuiFacetButton onClick={() => {pplServiceRequestor(upLogsQuery + filterVal + timeVal, "jsonData", setHistoryLogs)
-                                            setHistoryQuery(upLogsQuery + filterVal + timeVal)}}>
+            <EuiFacetButton
+              onClick={() => {
+                setHistoryQuery(upLogsQuery());
+              }}
+            >
               Up
             </EuiFacetButton>
-            <EuiFacetButton onClick={() => {pplServiceRequestor(downLogsQuery + filterVal + timeVal, "jsonData", setHistoryLogs)
-                                            setHistoryQuery(downLogsQuery + filterVal + timeVal)}}>
+            <EuiFacetButton
+              onClick={() => {
+                setHistoryQuery(downLogsQuery());
+              }}
+            >
               Down
             </EuiFacetButton>
           </EuiFacetGroup>
         </EuiFlexItem>
-      </EuiFlexGroup>
+      </>
       <EuiSpacer size="s" />
       <EuiPageContent>
-        <EuiBasicTable
+        <EuiInMemoryTable
```

Comment:
do we want to support pagination through backend queries?

Reply:
This depends on do we get pagination results from SQL/PPL or not? May be check how it is supported in the workbench

Reply:
i recently added `from` to PPL's `head` command, https://github.com/joshuali925/sql/blob/886579a716f2969516f4cdd8eb7584a841b98c35/docs/user/ppl/cmd/head.rst#example-3-get-first-n-results-after-offset-m

asking because loading items to memory can be ok for user created saved objects, but `historyLogs` might be too large

Reply:
@joshuali925 This is amazing 🥇